rust/sql/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#![allow(dead_code)]
+
+use arrow::datatypes::{Schema, SchemaRef};
+use arrow::error::Result;
+use arrow::record_batch::RecordBatch;
+
+// supported database modules
+pub mod postgres;
+
+///
+/// a SQL data source, used to read data from a SQL database into Arrow batches
+pub trait SqlDataSource {
```

Comment:
We should add something that identifies databases or their dialect. Could be an enum with what this lib supports, and then an `<enum>::Other(String)` for unknown databases, e.g. if someone implements something outside of this.

Reply:
This code seems similar to https://github.com/tokio-rs/rdbc (e.g. `Connection`, `Driver` and `ResultSet`)